modules/standard/Regex.chpl

```diff
@@ -611,12 +619,64 @@ record regex {
 
   proc init=(x: regex(?)) {
     this.exprType = x.exprType;
-    this.home = x.home;
-    this._regex = x._regex;
-    this.complete();
-    on home {
-      qio_regex_retain(_regex);
+    /* always bring the regex local */
+    this.home = here;
+    /* if it's local, retain and avoid the recompile (thread safe) */
+    if (x.home == here) {
+      this._regex = x._regex;
+      this.complete();
+      qio_regex_retain(x._regex);
+    } else {
+      /* otherwise recompile locally */
+      this.complete();
+      var serialized = x._serialize();
+      this._deserialize(serialized);
+    }
+  }
+
+  pragma "no doc"
+  proc _serialize() {
```

Comment:
Should there be a comment describing the serialization strategy, and why it's needed?

Reply:
Yeah that could be good. The strategy will be changing shortly and as its more complicated would warrant a comment on its own.